opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -259,6 +470,116 @@ impl OtlpEncoder {
+                                Ok(StatusCode::Ok) => true,
+                                Ok(StatusCode::Error) => false,
+                                _ => true, // Unset or unknown defaults to true
+                            }
+                        }
+                        None => true, // No status defaults to true
+                    };
+                    BondWriter::write_bool(&mut buffer, success);
+                }
+                FIELD_TRACE_ID => {
+                    let hex_bytes = Self::encode_id_to_hex::<32>(&span.trace_id);
+                    let hex_str = std::str::from_utf8(&hex_bytes).unwrap();
+                    BondWriter::write_string(&mut buffer, hex_str);
+                }
+                FIELD_SPAN_ID => {
+                    let hex_bytes = Self::encode_id_to_hex::<16>(&span.span_id);
+                    let hex_str = std::str::from_utf8(&hex_bytes).unwrap();
+                    BondWriter::write_string(&mut buffer, hex_str);
+                }
+                FIELD_TRACE_FLAGS => {
+                    BondWriter::write_numeric(&mut buffer, span.flags);
+                }
+                FIELD_NAME => {
+                    BondWriter::write_string(&mut buffer, &span.name);
+                }
+                FIELD_TRACE_STATE => {
+                    BondWriter::write_string(&mut buffer, &span.trace_state);
+                }
+                FIELD_PARENT_ID => {
+                    let hex_bytes = Self::encode_id_to_hex::<16>(&span.parent_span_id);
+                    let hex_str = std::str::from_utf8(&hex_bytes).unwrap();
+                    BondWriter::write_string(&mut buffer, hex_str);
+                }
+                FIELD_LINKS => {
+                    // Serialize links as JSON
+                    // TODO - consider more efficient serialization if needed
+                    let links_json = serde_json::to_string(
+                        &span
+                            .links
+                            .iter()
+                            .map(|link| {
+                                serde_json::json!({
+                                    "toSpanId": hex::encode(&link.span_id),
+                                    "toTraceId": hex::encode(&link.trace_id)
+                                })
+                            })
+                            .collect::<Vec<_>>(),
+                    )
+                    .unwrap_or_default();
+                    BondWriter::write_string(&mut buffer, &links_json);
+                }
+                FIELD_STATUS_MESSAGE => {
+                    if let Some(status) = &span.status {
+                        BondWriter::write_string(&mut buffer, &status.message);
+                    }
+                }
+                _ => {
+                    // Handle dynamic attributes
+                    if let Some(attr) = span.attributes.iter().find(|a| a.key == field.name) {
```

Comment:
Do we need to check if it's present in span attributes? Every item in `&[FieldDef]` input should be valid and serialized, right?